src/test/java/org/broadinstitute/hellbender/utils/recalibration/BQSRGathererUnitTest.java

```diff
@@ -0,0 +1,136 @@
+package org.broadinstitute.hellbender.utils.recalibration;
+
+import com.google.common.collect.ImmutableList;
+import org.broadinstitute.hellbender.utils.report.GATKReport;
+import org.broadinstitute.hellbender.utils.report.GATKReportTable;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public final class BQSRGathererUnitTest extends BaseTest {
+    private static final String testDir = BaseTest.publicTestDir + "/org/broadinstitute/hellbender/utils/recalibration/";
+    private static final File recal1 = new File(testDir + "HiSeq.1mb.1RG.sg1.table");
+    private static final File recal2 = new File(testDir + "HiSeq.1mb.1RG.sg2.table");
+    private static final File recal3 = new File(testDir + "HiSeq.1mb.1RG.sg3.table");
+    private static final File recal4 = new File(testDir + "HiSeq.1mb.1RG.sg4.table");
+    private static final File recal5 = new File(testDir + "HiSeq.1mb.1RG.sg5.table");
+
+    private static final ImmutableList<File> recalFiles = ImmutableList.of(recal1, recal2, recal3, recal4, recal5);
+
+    private static final File recalEmpty = new File(testDir + "HiSeq.1mb.1RG.empty.table");
+
+    //Note: the expected output was created by running GATK3, fixing table headers and sorting in unix "sort -k 2,2 -k 3,3 -k 4,4"
+    private static final File recal_original = new File(testDir + "HiSeq.1mb.1RG.noSG.table");
+
+    @DataProvider(name = "tables")
+    public Object[][] getTables(){
+        final List<File> filesWithEmpty = new ArrayList<>(recalFiles);
+        filesWithEmpty.add(recalEmpty);
+
+        return new Object[][] {
+                {Collections.nCopies(5, new File(testDir + "bqsr.manyObservations.piece.table")), new File(testDir + "bqsr.manyObservations.full.table")}, //gather files with multiple observations
+                {recalFiles, recal_original}, //gather several files
+                {filesWithEmpty, recal_original} //gather files with an empty table
+        };
+    }
+
+    @Test
+    public void resort(){
+        final GATKReport gatkReport = new GATKReport(recal_original);
+        gatkReport.print(System.out, GATKReportTable.Sorting.SORT_BY_COLUMN);
```

Comment:
This test should probably be removed,  I think I added it as part of my own manual testing.  It doesn't actually assert anything, so I don't think its useful.